acs-integration-tests/src/test/java/com/ge/predix/cloud/foundry/application/CloudFoundryApplicationHelper.java

```diff
@@ -0,0 +1,363 @@
+package com.ge.predix.cloud.foundry.application;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.MapUtils;
+import org.cloudfoundry.doppler.LogMessage;
+import org.cloudfoundry.operations.CloudFoundryOperations;
+import org.cloudfoundry.operations.applications.ApplicationEnvironments;
+import org.cloudfoundry.operations.applications.ApplicationHealthCheck;
+import org.cloudfoundry.operations.applications.DeleteApplicationRequest;
+import org.cloudfoundry.operations.applications.GetApplicationEnvironmentsRequest;
+import org.cloudfoundry.operations.applications.LogsRequest;
+import org.cloudfoundry.operations.applications.PushApplicationRequest;
+import org.cloudfoundry.operations.applications.SetEnvironmentVariableApplicationRequest;
+import org.cloudfoundry.operations.applications.StartApplicationRequest;
+import org.cloudfoundry.operations.services.BindServiceInstanceRequest;
+import org.cloudfoundry.operations.services.CreateServiceInstanceRequest;
+import org.cloudfoundry.operations.services.DeleteServiceInstanceRequest;
+import org.cloudfoundry.operations.services.GetServiceInstanceRequest;
+import org.cloudfoundry.operations.services.ServiceInstance;
+import org.cloudfoundry.operations.services.UnbindServiceInstanceRequest;
+import org.cloudfoundry.util.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.ge.predix.cloud.foundry.CloudFoundryService;
+import com.ge.predix.cloud.foundry.CloudFoundryUtilities;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Component
+final class CloudFoundryApplicationHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudFoundryApplicationHelper.class);
+    private static final int CF_SERVICE_INSTANCE_NAME_TAKEN = 60002;
+    private static final int MAX_RETRIES = 10;
+
+    @Autowired
+    private CloudFoundryOperations cloudFoundryOperations;
+
+    void pushApplication(final String applicationName, final Path application,
+            final Map<String, String> environmentVariables, final List<CloudFoundryService> services) {
+
+        pushApplication(this.cloudFoundryOperations, applicationName, application)
+                .thenMany(setEnvironmentVariables(this.cloudFoundryOperations, applicationName, environmentVariables))
+                .thenMany(createServiceInstances(this.cloudFoundryOperations, services))
```

Comment:
please check what happens when the service instance already exists. Should we delete instances first to ensure a clean setup for tests ?